operations/service_deploy_smoke_utils.go

```diff
@@ -325,38 +326,45 @@ func checkTaskLogContent(body []byte, mode agent.Mode) error {
 			return nil
 		}
 		// Validate that setup_group only runs in first task
-		if strings.Contains(page, "first") {
-			if !strings.Contains(page, "setup_group") {
+		const firstTaskGroupTaskLog = "smoke test is running first task in the task group"
+		const setupGroupLog = "smoke test is running the setup group"
+		if strings.Contains(page, firstTaskGroupTaskLog) {
+			if !strings.Contains(page, setupGroupLog) {
 				return errors.New("did not find setup_group in task logs for first task")
 			}
 		} else {
-			if strings.Contains(page, "setup_group") {
+			if strings.Contains(page, setupGroupLog) {
 				return errors.New("setup_group should only run in first task")
 			}
 		}
 
 		// Validate that setup_task and teardown_task run for all tasks
-		if !strings.Contains(page, "setup_task") {
+		const setupTaskLog = "smoke test is running the setup task"
+		if !strings.Contains(page, setupTaskLog) {
 			return errors.New("did not find setup_task in task logs")
 		}
+		const teardownTaskLog = "smoke test is running the teardown task"
```

Comment:
Nit: need to reference this on the line below as it's unused

Reply:
Good catch, fixed it.